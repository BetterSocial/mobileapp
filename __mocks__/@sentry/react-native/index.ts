export const addBreadcrumb = jest.fn();
