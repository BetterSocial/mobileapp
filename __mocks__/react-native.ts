import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN