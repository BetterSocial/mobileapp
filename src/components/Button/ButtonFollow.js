import React from 'react';
import {View, StyleSheet, TouchableNativeFeedback, Text} from 'react-native';
import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const ButtonFollow = ({handleSetFollow}) => (
  <TouchableNativeFeedback onPress={handleSetFollow}>
    <View style={styles.buttonFollow}>
      <Text style={styles.textButtonFollow}>Follow</Text>
    </View>
  </TouchableNativeFeedback>
);

const styles = StyleSheet.create({
  buttonFollow: {
    width: 58,
    height: 24,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 8,
    backgroundColor: COLORS.anon_primary
  },
  textButtonFollow: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 12,
    color: COLORS.white
  }
});

export default ButtonFollow;
