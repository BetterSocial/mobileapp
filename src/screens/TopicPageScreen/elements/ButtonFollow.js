import React from 'react';
import {View, StyleSheet, TouchableNativeFeedback, Text} from 'react-native';
import {fonts, normalize, normalizeFontSize} from '../../../utils/fonts';
import {COLORS} from '../../../utils/theme';

const ButtonFollow = ({handleSetFollow}) => {
  return (
    <TouchableNativeFeedback onPress={handleSetFollow}>
      <View style={styles.buttonFollow}>
        <Text style={styles.textButtonFollow}>Join</Text>
      </View>
    </TouchableNativeFeedback>
  );
};

const styles = StyleSheet.create({
  buttonFollow: {
    width: normalize(88),
    height: normalize(36),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 6,
    borderRadius: 8,
    backgroundColor: COLORS.blue
  },
  textButtonFollow: {
    fontFamily: fonts.inter[500],
    fontWeight: 'bold',
    fontSize: normalizeFontSize(12),
    color: COLORS.white,
    lineHeight: 24
  }
});

export default ButtonFollow;
