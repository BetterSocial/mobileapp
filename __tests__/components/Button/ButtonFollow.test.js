import * as React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import ButtonFollow from '../../../src/components/Button/ButtonFollow';

const mockNavigate = jest.fn()

jest.mock('@react-navigation/core', () => ({
  ...jest.requireActual('@react-navigation/core'),
  useNavigation: () => ({
    goBack: jest.fn(), navigate: mockNavigate 
  }),
}));


describe('ButtonFollow should run correctly', () => {

    afterEach(cleanup)
    it('should match snapshot', () => {
        const handleUnfollow = jest.fn()
        const {toJSON} = render(<ButtonFollow handleSetUnFollow={handleUnfollow} />)
        expect(toJSON).toMatchSnapshot()
    })
})