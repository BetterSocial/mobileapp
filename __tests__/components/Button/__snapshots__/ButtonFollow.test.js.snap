// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFollow should run correctly should match snapshot 1`] = `[Function]`;
